vendors/espressif/esp-idf/components/bt/bluedroid/bta/dm/bta_dm_co.c

```diff
@@ -361,14 +362,25 @@ void bta_dm_co_ble_set_rsp_key_req(UINT8 rsp_key)
 void bta_dm_co_ble_set_max_key_size(UINT8 ble_key_size)
 {
 #if (SMP_INCLUDED == TRUE)
-    if(ble_key_size >= BTM_BLE_MIN_KEY_SIZE && ble_key_size <= BTM_BLE_MAX_KEY_SIZE) {
+    if(ble_key_size >= bte_appl_cfg.ble_min_key_size && ble_key_size <= BTM_BLE_MAX_KEY_SIZE) {
         bte_appl_cfg.ble_max_key_size = ble_key_size;
     } else {
         APPL_TRACE_ERROR("%s error:Invalid key size value, key_size =%d",__func__, ble_key_size);
     }
 #endif  ///SMP_INCLUDED == TRUE
 }
 
+void bta_dm_co_ble_set_min_key_size(UINT8 ble_key_size)
+{
+#if (SMP_INCLUDED == TRUE)
+    if(ble_key_size >= BTM_BLE_MIN_KEY_SIZE && ble_key_size <= bte_appl_cfg.ble_max_key_size) {
```

Comment:
Is this check going to fail if bta_dm_co_ble_set_min_key_size() is called before bta_dm_co_ble_set_max_key_size() ?

Reply:
No, by default `ble_key_size <= bte_appl_cfg.ble_max_key_size` will be `BTM_BLE_MAX_KEY_SIZE` and `ble_key_size <= bte_appl_cfg.ble_min_key_size` will be `BTM_BLE_MIN_KEY_SIZE`.